### ISSUE TITLE
MMnist dataset for training settings T

### ISSUE THREAD

**Issue by 54wb** [Kay1794/MMVP-motion-matrix-based-video-prediction#6]
hi, thanks for your great work. I have noticed that you wrote in paper section 4.1 Dataset that in the training phase for MMNIST dataset to predict  the future 10 frames by the previous 10 frames
but in the forward function of model.py file, the input_image argument has shape [B, 20, 64, 64, 1], and this argument is passed down in the next all the way with a dimension of 20, and finally in the Future Composition phase it  becomes 10 dimensions.

    def forward(self, input_image,update_sim_matrix=True,inference=False):

        output_list = {}

        B, T, H, W, C = input_image.shape
        
        input_image = input_image.reshape(-1, H, W, C)  # B*T,H,W,C
        input_image = input_image.permute(0, 3, 1, 2)

It this in consistent with what the paper says? Or am I misunderstanding something? 
Thanks for your help!

**Comment by Kay1794** [comment]
Hi,

Thank you very much for your interest! Yes, the forward function carries the ground truth sequence for ground truth motion matrices calculation (can be used for visualization & debugging). However, the ground truth sequence is not needed during the inference time. Hope my answer makes sense to you. 

Best regards,
Yiqi

**Comment by 54wb** [comment]
hi, Thanks for your detail answer. But I have noticed that the future ground truth sequence have participated in the whole forward function not only visulization & debugging, so i want to know if i only use pre-sequence in forward, the metric would change?(Because i found when i train it with kth dataset, (16, 30, 1, 128, 128) input has terrible cuda memory used, I have 80G memory total but having OOT ) 
Do you have any idea? Thanks for your help

**Comment by Kay1794** [comment]
> hi, Thanks for your detail answer. But I have noticed that the future ground truth sequence have participated in the whole forward function not only visulization & debugging, so i want to know if i only use pre-sequence in forward, the metric would change?(Because i found when i train it with kth dataset, (16, 30, 1, 128, 128) input has terrible cuda memory used, I have 80G memory total but having OOT )
> Do you have any idea? Thanks for your help

For visualization, the forward function will also calculate the motion matrix of the gt sequences. It may be the reason for the extra gpu memory usage. You may want to  modify the code to avoid this and save the gpu memory. Don’t forget to modify the motion matrix visualization code in utils.py as well.
